### PR TITLE
Tests for ink contracts

### PR DESCRIPTION
**Pull request by sharma66mahesh**
Tests for ink contracts, to be run:
- on every commit/PR on main branch
- after every network upgrade

### CODE REVIEW

**Comment by sharma66mahesh** [comment]
Let's wait for [this PR](https://github.com/web3labs/ice-substrate/pull/277) to be merged.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -87,10 +87,21 @@ jobs:
...
           npm install -g yarn
-          cd ts-tests 
+          cd ts-tests/evm
           yarn install
           yarn run build
           yarn run test
+
+      - name: Run ink ts tests
+        env:
+          END_USER_WALLET_1_URI: ${{ secrets.TEST_INK_WALLET_SEED_1 }}
```
Comment (andrii-kl) [inline .github/workflows/test.yml:100]: I can't find where we use these seeds

Comment (andrii-kl) [reply .github/workflows/test.yml:100]: Do we really need 3 different wallets?

File: ts-tests/ink/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+- On server, set the env vars accordingly and run
+  ```sh
+  npm run test:server
+  ```
+- **Optional:** *Update `constants.ts` file for changing RPC endpoints*
+
+## Other Commands
+- Deploy a contract on mainnet to test state after upgrade
+  ```sh
+  npm run deploy-contract
```
Comment (andrii-kl) [inline ts-tests/ink/README.md:29]: Let's make in same way as we did for EVM tests. Add argument to run on different networks (arctic/stage/snow/)

File: ts-tests/ink/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+[![Author](https://img.shields.io/badge/author-@sharma66mahesh-blue.svg?style=flat-square)](https://github.com/sharma66mahesh)
+[![Maintainer](https://img.shields.io/badge/maintainer-@SNOWnetwork-blue.svg?style=flat-square)](https://github.com/web3labs/ice-substrate)
+[![Test Status](https://github.com/ibriz/ice-ink-contracts-tests/actions/workflows/mocha-tests.yml/badge.svg)
+</div>
+
+## How to Run
+- Tested with Node `v16.14.0`
+- Build contracts on `assets` folder.
+- **Optional:** *Update contract metadata and wasm paths on `constants.ts` file.* 
+- On local machine, create `.env.local` file with reference from `.env.sample` file and run
```
Comment (andrii-kl) [inline ts-tests/ink/README.md:16]: The first command should be `npm install`

**Comment by sharma66mahesh** [comment]
@andrii-kl if you are running locally, you need to have the `ice-node v0.4.58` executable available in your `ts-tests/ink/assets` folder

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -40,4 +40,52 @@ jobs:
...
+        run: |
+          npm install -g yarn
+          cd ts-tests/evm
+          yarn install
+          yarn run build
+          yarn run test
+
+      - name: Run ink ts tests
+        env:
+          TEST_INK_WALLET_SEED_1: ${{ secrets.TEST_INK_WALLET_SEED_1 }}
```
Comment (andrii-kl) [inline .github/workflows/build.yml:85]: I can't find where we use these seeds in the project

**Review by andrii-kl — APPROVED** [review]

**Pull request merged**